### ISSUE TITLE
jax.mask works on jnp.sum(x) but not on jax.nn.logsumexp(x)

### ISSUE THREAD

**Issue by sokrypton** [google/jax#6393]
jax.mask is really cool! but it seems like it does not work for logsumexp() :(

colab example:
https://colab.research.google.com/drive/1HXp3QDIRwzWjXbOW9UDoo1-UV5U4MX74

**Comment by mattjj** [comment]
Thanks for the question, and the positive words about `jax.mask`!

`jax.mask` is a prototype, and instead of developing it further we're working on a follow-up which should provide all its capabilities plus more. While I'm guessing that we could make `jax.mask` work with `logsumexp` in most cases without too much trouble, you might just run into another issue with it. For that reason I'm inclined to think we should focus our efforts on the new thing.

What do you think?

**Comment by sokrypton** [comment]
I can wait for the follow-up! (any preview?)

**Comment by hawkinsp** [comment]
We've dropped `jax.mask` from JAX, but keep your eyes out for better dynamic shape support in JAX soon!
